fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java

```diff
@@ -1397,4 +1427,31 @@ public RecordLayerPropertyStorage getPropertyStorage() {
         return config.getPropertyStorage();
     }
 
+    /**
+     * Get key ranges that encountered conflicts when trying to {@link #commit} failed due to {@code NOT_COMMITTED}.
+     * @return list of conflict ranges
+     */
+    @Nullable
+    public List<Range> getNotCommittedConflictingKeys() {
+        return notCommittedConflictingKeys;
+    }
+
+    @SuppressWarnings("PMD.CloseResource")
+    private CompletableFuture<Void> readConflictingKeys() {
+        notCommittedConflictingKeys = new ArrayList<>();
```

Comment:
It seems a bit strange that this sets the instance variable at the beginning of the method and then updates the instance variable asynchronously. It feels like this should have a local variable, and then when the async callback completes, ether it returns the list (and `commit()` sets the instance variable) or the instance variable gets set within this method. That would then mean that the `notCommittedConflictingKeys` variable would either be `null` or would contain a complete set of conflicts (rather than certain incorrect usage resulting in the viewer getting a non-`null` but incomplete set of conflicts).

Reply:
I switched it to a value-returning async method instead of that strange side-effecting one.